tests/llmobs/test_llmobs_span_writer.py

```diff
@@ -7,92 +6,123 @@
+            1,
+            403,
+            b'{"errors":[{"status":"403","title":"Forbidden","detail":"API key is invalid"}]}',
+        )
 
 
 @pytest.mark.vcr_logs
 def test_send_timed_events(mock_writer_logs):
-    llmobs_span_writer = LLMObsSpanWriter(site="datad0g.com", api_key=dd_api_key, interval=0.01, timeout=1)
-    llmobs_span_writer.start()
-    mock_writer_logs.reset_mock()
-
-    llmobs_span_writer.enqueue(_completion_event())
-    time.sleep(0.1)
-    mock_writer_logs.debug.assert_has_calls([mock.call("sent %d LLMObs %s events to %s", 1, "span", INTAKE_ENDPOINT)])
-    mock_writer_logs.reset_mock()
-    llmobs_span_writer.enqueue(_chat_completion_event())
-    time.sleep(0.1)
-    mock_writer_logs.debug.assert_has_calls([mock.call("sent %d LLMObs %s events to %s", 1, "span", INTAKE_ENDPOINT)])
+    with override_global_config(
+        dict(
+            _dd_site=DATADOG_SITE,
+            _dd_api_key="foobar.baz",
+        )
+    ):
+        llmobs_span_writer = LLMObsSpanWriter(is_agentless=True, interval=0.01, timeout=1)
+        llmobs_span_writer.start()
+        mock_writer_logs.reset_mock()
+
+        llmobs_span_writer.enqueue(_completion_event())
+        time.sleep(0.1)
+        mock_writer_logs.debug.assert_has_calls([mock.call("encode %d LLMObs span events to be sent", 1)])
+        mock_writer_logs.reset_mock()
+        llmobs_span_writer.enqueue(_chat_completion_event())
+        time.sleep(0.1)
+        mock_writer_logs.debug.assert_has_calls([mock.call("encode %d LLMObs span events to be sent", 1)])
 
 
 @pytest.mark.vcr_logs
 def test_send_multiple_events(mock_writer_logs):
-    llmobs_span_writer = LLMObsSpanWriter(site="datad0g.com", api_key=dd_api_key, interval=0.01, timeout=1)
-    llmobs_span_writer.start()
-    mock_writer_logs.reset_mock()
-
-    llmobs_span_writer.enqueue(_completion_event())
-    llmobs_span_writer.enqueue(_chat_completion_event())
-    time.sleep(0.1)
-    mock_writer_logs.debug.assert_has_calls([mock.call("sent %d LLMObs %s events to %s", 2, "span", INTAKE_ENDPOINT)])
+    with override_global_config(
+        dict(
+            _dd_site=DATADOG_SITE,
+            _dd_api_key="foobar.baz",
+        )
+    ):
+        llmobs_span_writer = LLMObsSpanWriter(is_agentless=True, interval=0.01, timeout=1)
+        llmobs_span_writer.start()
+        mock_writer_logs.reset_mock()
+
+        llmobs_span_writer.enqueue(_completion_event())
+        llmobs_span_writer.enqueue(_chat_completion_event())
+        time.sleep(0.1)
+        mock_writer_logs.debug.assert_has_calls([mock.call("encode %d LLMObs span events to be sent", 2)])
 
 
 def test_send_on_exit(mock_writer_logs, run_python_code_in_subprocess):
-    out, err, status, pid = run_python_code_in_subprocess(
-        """
+    with override_global_config(
+        dict(
+            _dd_site=DATADOG_SITE,
+            _dd_api_key="foobar.baz",
+        )
+    ):
```

Comment:
We can't override the global config here as that doesn't reach the subprocess, here we should modify the env used in the subprocess test like [here](https://github.com/DataDog/dd-trace-py/blob/main/tests/contrib/openai/test_openai_v1.py#L1203-L1229).
Note that the env just needs to include the `"PYTHONPATH"`, you can replace `OPENAI_API_KEY/DD_OPENAI_METRICS_ENABLED` in that example with `DD_API_KEY/DD_SITE`.